### PR TITLE
chore(agw): address compiler warnings

### PR DESCRIPTION
**Pull request by fyt000**
## Summary
Address #10384 
- casting to uint64_t for comparison
- print address of `pcap_dispatch` (unless this wasn't the intended behaviour)
- catch exception by `const&`
- remove unneeded variables
- (void) to surpress unused warnings

## Test Plan
Triggering a full rebuild - only see warning from external sources or protos

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/li_agent/src/main.cpp [hunk 1]
```diff
@@ -116,7 +116,7 @@ int main(void) {
     return -1;
   }
 
-  MLOG(MERROR) << "CleanUP " << pcap_dispatch;
+  MLOG(MERROR) << "CleanUP " << reinterpret_cast<void*>(&pcap_dispatch);
```
Comment (themarwhal) [inline lte/gateway/c/li_agent/src/main.cpp]: Not sure why this is at error log level @koolzz should we just remove this log? Or bump it down to debug.

Comment (koolzz) [reply lte/gateway/c/li_agent/src/main.cpp]: Looks to me like we should, tagging @ymasmoudi  for visibility

File: lte/gateway/c/sctpd/src/sctp_connection.cpp [hunk 2]
```diff
@@ -200,7 +200,7 @@ SctpStatus SctpConnection::HandleClientSock(int sd) {
     SctpAssoc&& assoc = SctpAssoc();
     try {
       assoc = _sctp_desc.getAssoc(sinfo.sinfo_assoc_id);
-    } catch (std::out_of_range) {
+    } catch (const std::out_of_range&) {
```
Comment (ardzoht) [inline lte/gateway/c/sctpd/src/sctp_connection.cpp:203]: 👍

File: orc8r/gateway/c/common/logging/magma_logging_init.h [hunk 3]
```diff
@@ -31,6 +31,7 @@ static void set_verbosity(uint32_t verbosity) {
 
 // get_verbosity gets the the global logging verbosity
 static google::int32 get_verbosity() {
+  (void)get_verbosity;
```
Comment (ardzoht) [inline orc8r/gateway/c/common/logging/magma_logging_init.h]:
```suggestion
  (void)get_verbosity; // casting to void to suppress unused reference warning
```

Comment (ardzoht) [reply orc8r/gateway/c/common/logging/magma_logging_init.h]: you can also define a macro for this:

```#define SUPPRESS_WARNING(a) (void)a```

File: orc8r/gateway/c/common/logging/magma_logging_init.h [hunk 4]
```diff
@@ -49,12 +50,17 @@ static void init_logging(const char* service_name) {
 #include <iostream>
 
 namespace magma {
-static void set_verbosity(__attribute__((unused)) uint32_t verbosity) {}
+static void set_verbosity(__attribute__((unused)) uint32_t verbosity) {
+  (void)set_verbosity;
```
Comment (ardzoht) [inline orc8r/gateway/c/common/logging/magma_logging_init.h]: add comment here as well

File: orc8r/gateway/c/common/logging/magma_logging_init.h [hunk 5]
```diff
@@ -49,12 +50,17 @@ static void init_logging(const char* service_name) {
...
+}
 // get_verbosity gets the the global logging verbosity
 static uint32_t get_verbosity() {
+  (void)get_verbosity;
```
Comment (ardzoht) [inline orc8r/gateway/c/common/logging/magma_logging_init.h]: add comment

File: orc8r/gateway/c/common/logging/magma_logging_init.h [hunk 6]
```diff
@@ -49,12 +50,17 @@ static void init_logging(const char* service_name) {
...
   return 0;
 }
-static void init_logging(__attribute__((unused)) const char* service_name) {}
+static void init_logging(__attribute__((unused)) const char* service_name) {
+  (void)init_logging;
```
Comment (ardzoht) [inline orc8r/gateway/c/common/logging/magma_logging_init.h]: add comment for suppressing warning

**Review by ssanadhya — APPROVED** [review]
Changes to c/sctpd/* look good to me!

**Comment by fyt000** [comment]
Also adding destructor to the `ProxyConnectorImpl` https://github.com/magma/magma/issues/10384#issuecomment-976891835

Let me know what is the best way to test it, thanks!

**Review by themarwhal — APPROVED** [review]

**Comment by fyt000** [comment]
I am not authorized to merge, if it looks good please merge it for me thanks!

**Pull request merged**